packages/aws-cdk/lib/contextplugins.ts

```diff
@@ -48,21 +53,85 @@ export class SSMContextProviderPlugin implements ContextProviderPlugin {
   }
 }
 
+export class HostedZoneContextProviderPlugin implements ContextProviderPlugin {
+
+  constructor(private readonly aws: SDK) {
+  }
+
+  public async getValue(filter: cxapi.ContextProviderProps) {
+    const account = filter.account;
+    const region = filter.region;
+    if (!this.isHostedZoneProps(filter.props)) {
+      throw new Error(`HostedZoneProvider requires domainName property to be set in ${filter.props}`);
+    }
+    const props: cxapi.HostedZoneProviderProps = filter.props;
+    const domainName = props.domainName;
+    const privateZone: boolean = !!props.privateZone;
+    const vpcId =
```

Comment:
??

Reply:
yeah -- this looks like a coding on a plane bonus bug ✈️ 🐛 

Should we expand the scope of this PR and find a away to integration test these?

Reply:
I'm not opposed to it, but I have no clever ideas on how to do it that wouldn't take a long time to build and/or make tests complicated to run.

For example, if we're going to test the actual service call, we'd need AWS credentials and network access on every build--something we can (mostly) avoid now.

You might potentially consider mocking/stubbing the external APIs, OR having a test that you run manually, and a signature (like a hash of the file) to assert that you manually ran the test on the latest version of the source.

Reply:
I haven't fully looked at this yet, but perhaps you can tell me I'm walking into a big thing.

My thought was to keep this as simple as possible. Enable cdk-integ to be able to run more than one stack. Create resources in the first stack and use providers to find those resources in the second stack. Then sanitize the cdk.json file with the appropriate region/account information such that AWS API credentials are not required after the developer runs the integration test locally.

This definitely isn't simple, but would that work and not be a nightmare to implement?

Reply:
It's definitely a bit out of the mold of our current integ tests.

* You can't comfortable write this test in `aws-cdk` package (I don't think), it has to go in one of the library packages. Which one?
* How are we going to specify the order between multiple stacks in `cdk-integ`? We have explicitly punted on that because there's no way to automatically infer it, and no place to specify it.
* We don't check in `cdk.json` usually, and in fact `cdk-integ` automatically deletes it. To be able to run an offline test we should keep that file; but preferably not all files in all cases, so only *some* instances of that file...